File: app/src/main/java/com/siju/acexplorer/helper/DateUtils.java
package com.siju.acexplorer.helper;

import java.util.Calendar;

public class DateUtils {

    public static boolean isYesterday(long ms) {
        Calendar now = Calendar.getInstance();
        Calendar cdate = Calendar.getInstance();
        cdate.setTimeInMillis(ms);

        now.add(Calendar.DATE,-1);

        return now.get(Calendar.YEAR) == cdate.get(Calendar.YEAR)
                && now.get(Calendar.MONTH) == cdate.get(Calendar.MONTH)
                && now.get(Calendar.DATE) == cdate.get(Calendar.DATE);
    }

    public static boolean isToday(long ms) {
        Calendar now = Calendar.getInstance();
        Calendar cdate = Calendar.getInstance();
        cdate.setTimeInMillis(ms);

        now.add(Calendar.DATE,  0);

        return now.get(Calendar.YEAR) == cdate.get(Calendar.YEAR)
                && now.get(Calendar.MONTH) == cdate.get(Calendar.MONTH)
                && now.get(Calendar.DATE) == cdate.get(Calendar.DATE);
    }

    public static boolean isThisWeek(long ms) {
        Calendar now = Calendar.getInstance();
        Calendar cdate = Calendar.getInstance();
        cdate.setTimeInMillis(ms);


        return now.get(Calendar.YEAR) == cdate.get(Calendar.YEAR)
                && now.get(Calendar.MONTH) == cdate.get(Calendar.MONTH)
                && now.get(Calendar.WEEK_OF_YEAR) == cdate.get(Calendar.WEEK_OF_YEAR);
    }

    public static boolean isThisMonth(long ms) {
        Calendar now = Calendar.getInstance();
        Calendar cdate = Calendar.getInstance();
        cdate.setTimeInMillis(ms);


        return now.get(Calendar.YEAR) == cdate.get(Calendar.YEAR)
                && now.get(Calendar.MONTH) == cdate.get(Calendar.MONTH);
    }
}

File: app/src/main/java/com/siju/acexplorer/helper/DateUtils.kt
package com.siju.acexplorer.helper

import java.text.SimpleDateFormat
import java.util.*

private const val DATE_FORMAT = "MMM yyyy"
private val dateFormat = SimpleDateFormat(DATE_FORMAT, Locale.ENGLISH)
object DateUtils {

    fun isToday(ms: Long): Boolean {
        val now = Calendar.getInstance()
        val date = Calendar.getInstance()
        date.timeInMillis = ms
        return now[Calendar.YEAR] == date[Calendar.YEAR] && now[Calendar.MONTH] == date[Calendar.MONTH]
                && now[Calendar.DATE] == date[Calendar.DATE]
    }

    fun isYesterday(ms: Long): Boolean {
        val yestDate = Calendar.getInstance()
        yestDate.add(Calendar.DATE, -1)
        val inputDate = Calendar.getInstance()
        inputDate.timeInMillis = ms
        return yestDate.get(Calendar.DATE) == inputDate.get(Calendar.DATE)
    }

    fun isThisWeek(ms: Long): Boolean {
        val now = Calendar.getInstance()
        val inputDate = Calendar.getInstance()
        inputDate.timeInMillis = ms
        return now.get(Calendar.WEEK_OF_YEAR) == inputDate.get(Calendar.WEEK_OF_YEAR)
    }

    fun isThisMonth(ms: Long): Boolean {
        val now = Calendar.getInstance()
        val date = Calendar.getInstance()
        date.timeInMillis = ms
        return (now[Calendar.YEAR] == date[Calendar.YEAR]
                && now[Calendar.MONTH] == date[Calendar.MONTH])
    }

    fun getMonthYear(ms : Long) : String {
        val date = Date(ms)
        return dateFormat.format(date)
    }

}